vip-api/src/main/java/fr/insalyon/creatis/vip/api/business/StatsApiBusiness.java

```diff
@@ -0,0 +1,229 @@
+import fr.insalyon.creatis.vip.core.client.bean.User;
+import fr.insalyon.creatis.vip.core.client.view.util.CountryCode;
+import fr.insalyon.creatis.vip.core.server.business.BusinessException;
+import fr.insalyon.creatis.vip.core.server.business.StatsBusiness;
+import fr.insalyon.creatis.vip.core.server.business.StatsBusiness.UserSearchCriteria;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.Month;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.List;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+@Service
+public class StatsApiBusiness {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private final StatsBusiness statsBusiness;
+    private final Supplier<Connection> connectionSupplier;
+
+    @Autowired
+    public StatsApiBusiness(
+            StatsBusiness statsBusiness,
+            Supplier<Connection> connectionSupplier) {
+        this.statsBusiness = statsBusiness;
+        this.connectionSupplier = connectionSupplier;
+    }
+
+    public UsersNumber getUsersRegisteredNumber(
+            @Nullable String startDateString,
+            @Nullable String endDateString) throws ApiException {
+        // build search criteria
+        UserSearchCriteria searchCriteria =
+                new UserSearchCriteria()
+                .withRegistrationStart(parseDate(startDateString))
+                .withRegistrationEnd(parseDate(endDateString));
+
+        // do search
+        Long usersRegisteredNumber;
+        try(Connection connection = connectionSupplier.get()) {
+            usersRegisteredNumber =
+                    statsBusiness.getUsersRegisteredNumber(
+                            searchCriteria, connection);
+        } catch (SQLException ex) {
+            logger.error("Error handling a connection", ex);
+            throw new ApiException(ex);
+        } catch (SQLRuntimeException | BusinessException ex) {
+            throw new ApiException(ex);
+        }
+        // build response object
+        LocalDate startDate = searchCriteria.getRegistrationStart();
+        if (startDate == null) {
+            startDate = LocalDate.of(2010, Month.JANUARY, 1);
+        }
+        LocalDateTime endDateTime = LocalDateTime.now();
+        if (endDateString != null) {
+            endDateTime = searchCriteria.getRegistrationEnd().atStartOfDay();
+        }
+        return new UsersNumber(startDate.atStartOfDay(), endDateTime, usersRegisteredNumber);
+    }
+
+
+    public UsersList getAllUsers() throws ApiException {
+        return getUsersList(new UserSearchCriteria());
+    }
+
+    public UsersList getAllUsersBetweenDate(
```

Comment:
Any particular reason from not calling this "getAllUsersFromDate" ?

Reply:
Copy/paste errror....